### ISSUE TITLE
Bug: [no-useless-default-assignment] false positive with ternary

### ISSUE THREAD

**Issue by ulrichstark** [typescript-eslint/typescript-eslint#11850]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.9.2&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3gIyUiAXDnhCFCgNwBQy6WA3vFPALzwDkoAZlAK4QZ28AL6ti%2BeAH54jKEXYAJEBAhIhoovWEUgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS0AJogBmAQwR1xAygHMmAW0Qt0URNGgdokcGAC%2BIfUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const bool: boolean;
const { a = 'default' } = bool ? { a: 'Hello' } : {};
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-useless-default-assignment": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Don't report `= 'default'`

### Actual Result

Default value is useless because the property is not optional

### Additional Info

Originally reported by @jimmy-zhening-luo in https://github.com/typescript-eslint/typescript-eslint/issues/11847#issuecomment-3659098774.

**Comment by Mister-Hope** [comment]
I am also meeting this issue, a [repro](https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=FAEwpgxgNghgTmABBA9gOwM4BdECsMBCUKEA1gFyLZwCWaA5gNoC6A3MKpjo-ogLyIARIIA0eDABkYDfkMHNZ%2BIiVLBEiAPyIehYmUYAWZgDoEAB1gQwAQShQAFAHoAegB00AH3cASR-THCAJRiSnqkjADMzMxqiJQs7MCOjogAZjQAHmAgiAgYAK5QWGIA7gAWNFBIvNI5vLAyJZVQiABGSNR09IgeiPlo4Olo2Ukp5TQQZYhNdm0IMKSIsFhYYHDIKODTKHCkXYgwGNNI8EhgGWaQWPtYZWAAtohYKG0dWLQMHOjY2vgATCFJNJ6H9EAoBKEVLEtDplPojKYwBYYFZbA4XO4vGhfP45MFxHDwlEYup4mwgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS0AJogBmAQwR1xAygHMmAW0Qt0URNGgdokcGAC%2BIfUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false) with array here.

**Comment by ulrichstark** [comment]
> I am also meeting this issue, a [repro](https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=FAEwpgxgNghgTmABBA9gOwM4BdECsMBCUKEA1gFyLZwCWaA5gNoC6A3MKpjo-ogLyIARIIA0eDABkYDfkMHNZ%2BIiVLBEiAPyIehYmUYAWZgDoEAB1gQwAQShQAFAHoAegB00AH3cASR-THCAJRiSnqkjADMzMxqiJQs7MCOjogAZjQAHmAgiAgYAK5QWGIA7gAWNFBIvNI5vLAyJZVQiABGSNR09IgeiPlo4Olo2Ukp5TQQZYhNdm0IMKSIsFhYYHDIKODTKHCkXYgwGNNI8EhgGWaQWPtYZWAAtohYKG0dWLQMHOjY2vgATCFJNJ6H9EAoBKEVLEtDplPojKYwBYYFZbA4XO4vGhfP45MFxHDwlEYup4mwgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS0AJogBmAQwR1xAygHMmAW0Qt0URNGgdokcGAC%2BIfUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false) with array here.

This is already covered by #11845 and fixed if that PR is merged & released.
